src/approaches/grammar_search_invention_approach.py

```diff
@@ -34,32 +34,33 @@ def _create_grammar(dataset: Dataset, given_predicates: Set[Predicate]
     # We start with the given predicates because we want to allow
     # negated and quantified versions of the given predicates, in
     # addition to negated and quantified versions of new predicates.
-    given_grammar = _GivenPredicateGrammar(given_predicates)
+    # given_grammar = _GivenPredicateGrammar(given_predicates)
     # Next, we consider various ways to split single feature values
     # across our dataset.
     sfi_grammar = _SingleFeatureInequalitiesPredicateGrammar(dataset)
     # This chained grammar has the effect of enumerating first the
     # given predicates, then the single feature inequality ones.
-    chained_grammar = _ChainPredicateGrammar([given_grammar, sfi_grammar])
+    # chained_grammar = _ChainPredicateGrammar([given_grammar, sfi_grammar])
     # Now, the chained grammar will undergo a series of transformations.
     # For each predicate enumerated by the chained grammar, we also
     # enumerate the negation of that predicate.
-    negated_grammar = _NegationPredicateGrammarWrapper(chained_grammar)
+    negated_grammar = _NegationPredicateGrammarWrapper(sfi_grammar)
     # For each predicate enumerated, we also enumerate foralls for
     # that predicate, along with appropriate negations.
-    forall_grammar = _ForallPredicateGrammarWrapper(negated_grammar)
+    # forall_grammar = _ForallPredicateGrammarWrapper(negated_grammar)
```

Comment:
do you want to go back to the multiple grammar names?

Reply:
i think it'll be better to just have flags like `include_givens_in_grammar` and `include_forall_in_grammar`